plugins/qeta-backend/src/service/upload/attachmentStorageEngine.ts

```diff
@@ -0,0 +1,16 @@
+import { Attachment } from '@drodil/backstage-plugin-qeta-common';
+import { File } from '../types';
+import { Config } from '@backstage/config/index';
+import { QetaStore } from '../../database/QetaStore';
+
+export type AttachmentStorageEngineOptions = {
+    config: Config;
+    database: QetaStore;
+};
+  
+
+export interface AttachmentStorageEngine {
+    handleFile: (file: File, options?: { postId?: number; answerId?: number; collectionId?: number }) => Promise<Attachment>;
```

Comment:
Just a nit-pick, this uses File while other methods use Attachment. Maybe rename either one?

Reply:
I kept this as is since the original implementation also used `File` here as the input and returning an `Attachment`. I don't know if that was on purpose. Looks like `File` only represents a physical file, while `Attachment` adds some additional metadata on top of that. I think I can do everything I need to with `Attachment` but didn't want to break things unnecessarily ;).

Reply:
I've looked into this, but I don't think it is worth it at this point to be honest. `Attachment` is missing some crucial information that the storage engines will need, such as file size, mime type and extension.